### PR TITLE
DOC: Fix version on RTD stable

### LINKED ISSUE #11673: Stable docs have unstable version number
https://docs.astropy.org/en/stable/ appears to have a development version number (see top left). It should have a stable version number. Interestingly we're also seeing the same issue over at sunpy (https://github.com/sunpy/sunpy/issues/5225)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to fix  #11673

I am not milestoning this because it should not be ported to any branches. This is a special temporary hack.

### CODE REVIEW

**Review by eteq — APPROVED** [review]
LGTM assuming RTD passes!

**Comment by eteq** [comment]
On one hand :cry: that the build failed.  On the other, I found this amusing in the sphinx log...:
```
    used = env._viewcode_modules[modname][2]  # type: ignore
TypeError: 'bool' object is not subscriptable
```
I guess ``type:ignore`` shouldn't have been there :wink:

**Comment by pllim** [comment]
Oh... Yeah, should probably backport #11724 here for now.

**Comment by pllim** [comment]
https://astropy--11734.org.readthedocs.build/en/11734/ shows 4.2.1

File: setup.cfg [hunk 1]
```diff
@@ -79,7 +79,7 @@ all =
     ipython
     pytest
 docs =
-    sphinx
+    sphinx<4
```
Comment (pllim) [inline setup.cfg:82]: I made #11763 to unpin in dev, but I don't think we need to do it here as the stable branch will get overwritten when 4.3 is out anyway.

**Pull request merged**